src/runtime/contrib/tensorrt/tensorrt_module.cc

```diff
@@ -197,7 +200,19 @@ class TensorRTModule : public runtime::ModuleNode {
       if (!runtime::TypeMatch(arg->dtype, kDLFloat, 32)) {
         LOG(FATAL) << "Only float32 inputs are supported.";
       }
-      bindings[binding_index] = reinterpret_cast<float*>(arg->data);
+      if (inputs[i]->ctx.device_type == kDLCPU) {
+        if (device_pointers[binding_index] == nullptr) {
+          void* gpu_mem;
+          TVMDeviceAllocDataSpace(gpu_ctx, tvm::runtime::GetDataSize(*inputs[i]), 256,
```

Comment:
We can pass `&device_pointers[binding_index]` to `TVMDeviceAllocDataSpace` directly instead of using `gpu_mem`.

Same for the output buffers

Reply:
agreed. I made that change to remove gpu_mem.
I added code to free device buffers in ~TensorRTModule()